packages/cursorless-engine/src/generateSpokenForm/generateSpokenForm.ts

```diff
@@ -0,0 +1,203 @@
+        targetToSpokenForm(action.target1),
+        connectives.swapConnective,
+        targetToSpokenForm(action.target2),
+      ];
+
+    case "callAsFunction":
+      if (action.argument != null) {
+        throw new NoSpokenFormError(`Action '${action.name}' with argument`);
+      }
+      return [
+        actions[action.name],
+        targetToSpokenForm(action.callee),
+        // targetToSpokenForm(action.argument),
+      ];
+
+    case "wrapWithPairedDelimiter":
+    case "rewrapWithPairedDelimiter":
+      return [
+        surroundingPairDelimitersToSpokenForm(action.left, action.right),
+        actions[action.name],
+        targetToSpokenForm(action.target),
+      ];
+
+    case "pasteFromClipboard":
+      return [
+        actions[action.name],
+        destinationToSpokenForm(action.destination),
+      ];
+
+    case "insertSnippet":
+      return [
+        actions[action.name],
+        insertionSnippetToSpokenForm(action.snippetDescription),
+        destinationToSpokenForm(action.destination),
+      ];
+
+    case "generateSnippet":
+      if (action.snippetName != null) {
+        throw new NoSpokenFormError(`${action.name}.snippetName`);
+      }
+      return [actions[action.name], targetToSpokenForm(action.target)];
+
+    case "wrapWithSnippet":
+      return [
+        wrapperSnippetToSpokenForm(action.snippetDescription),
+        actions[action.name],
+        targetToSpokenForm(action.target),
+      ];
+
+    case "highlight": {
+      if (action.highlightId != null) {
+        throw new NoSpokenFormError(`${action.name}.highlightId`);
+      }
+      return [actions[action.name], targetToSpokenForm(action.target)];
+    }
+
+    default: {
+      return [actions[action.name], targetToSpokenForm(action.target)];
+    }
+  }
+}
+
+function targetToSpokenForm(
+  target: PartialTargetDescriptor,
+): RecursiveArray<string> {
+  switch (target.type) {
+    case "list":
+      if (target.elements.length < 2) {
+        throw new NoSpokenFormError("List target with < 2 elements");
+      }
+
+      return target.elements.map((element, i) =>
+        i === 0
+          ? targetToSpokenForm(element)
+          : [connectives.listConnective, targetToSpokenForm(element)],
+      );
+
+    case "range": {
+      const anchor = targetToSpokenForm(target.anchor);
+      const active = targetToSpokenForm(target.active);
+      const connective = getRangeConnective(
+        target.excludeAnchor,
+        target.excludeActive,
+        target.rangeType,
+      );
+      return [anchor, connective, active];
+    }
+
+    case "primitive":
+      return primitiveTargetToSpokenForm(target);
+
+    case "implicit":
+      return [];
+  }
+}
+
+function destinationToSpokenForm(
+  destination: DestinationDescriptor,
+): RecursiveArray<string> {
+  switch (destination.type) {
+    case "list":
+      if (destination.destinations.length < 2) {
+        throw new NoSpokenFormError("List destination with < 2 elements");
+      }
```

Comment:
No way to speak this one

Reply:
Same here. I would argue that this should be a normal error.